backend/src/services/release.ts

```diff
@@ -295,6 +335,142 @@ export async function getReleaseBySemver(user: UserInterface, modelId: string, s
   return release
 }
 
+function getSemverQueryBounds(querySemver: string) {
+  const semverRangeStandardised = semver.validRange(querySemver, { includePrerelease: false })
+
+  if (!semverRangeStandardised) {
+    throw BadReq(`Semver range, '${querySemver}' is invalid.`)
+  }
+
+  const [expressionA, expressionB] = semverRangeStandardised.split(' ')
+
+  let lowerInclusivity, lowerSemver, upperInclusivity, upperSemver
+
+  //If lower semver
+  if (expressionA.includes('>')) {
+    lowerInclusivity = expressionA.includes('>=')
+    lowerSemver = expressionA.replace(/[<>=]/g, '')
+  } else {
+    //upper semver
+    upperInclusivity = expressionA.includes('<=')
+    upperSemver = expressionA.replace(/[<=]/g, '')
+  }
+
+  if (expressionB) {
+    upperInclusivity = expressionB.includes('<=')
+    upperSemver = expressionB.replace(/[<=]/g, '')
+  }
+
+  let lowerSemverObj, upperSemverObj
+  if (lowerSemver) {
+    lowerSemverObj = semverStringToObject(lowerSemver)
+  }
+
+  if (upperSemver) {
+    upperSemverObj = semverStringToObject(upperSemver)
+  }
+
+  return { lowerSemverObj, upperSemverObj, lowerInclusivity, upperInclusivity }
+}
+
+interface QueryBoundInterface {
+  $or: (
+    | {
+        'semver.major':
+          | {
+              $lte: number
+            }
+          | {
+              $gte: number
+            }
+        'semver.minor':
+          | {
+              $lte: number
+            }
+          | { $gte: number }
+        'semver.patch':
+          | {
+              $gte: number
+            }
+          | { $gt: number }
+          | { $lte: number }
+          | { $lt: number }
+      }
+    | {
+        'semver.major':
+          | {
+              $gt: number
+            }
+          | { $lt: number }
+      }
+    | {
+        'semver.major':
+          | {
+              $gte: number
+            }
+          | { $lte: number }
+        'semver.minor':
+          | {
+              $gt: number
+            }
+          | { $lt: number }
+      }
+  )[]
+}
+
+export function getQuerySyntax(querySemver: string, modelID: string) {
```

Comment:
We shouldn't need to export this? To make it clear what this function is doing, could we change the name to something like `convertSemverQueryToMongoQuery()`

Reply:
I have changed the name. I have exported as I test it in release.spec.ts test file.